extensions/devdocs/src/index.tsx

```diff
@@ -0,0 +1,199 @@
+        await setState({data: JSON.parse(text.toString()), isLoading: true});
+      }
+
+      // Refresh Cache
+      try {
+        const data = await fetch(url).then((res) => res.json());
+        await setState({data: data as T, isLoading: false});
+        await writeFile(cachePath, JSON.stringify(data));
+      } catch (error) {
+        console.error(error)
+        showToast(ToastStyle.Failure, "Could not refresh cache!", "Please Check your connexion");
+      }
+    }
+    fetchWithCache();
+  }, [url]);
+
+  return state;
+}
+
+export default function DocList(): JSX.Element {
+  const { data, isLoading } = useFetchWithCache<Doc[]>(`${DEVDOCS_BASE_URL}/docs/docs.json`, "index.json");
+  const { docs: visitedDocs, visitDoc } = useVisitedDocs();
+
+  return (
+    <List isLoading={(!data && !data) || isLoading}>
+      <List.Section title="Last Visited">
+        {visitedDocs?.map((doc) => (
+          <DocItem key={doc.slug} doc={doc} onVisit={() => visitDoc(doc)} />
+        ))}
+      </List.Section>
+      <List.Section title="All">
+        {data?.map((doc) => (
+          <DocItem key={doc.slug} doc={doc} onVisit={() => visitDoc(doc)} />
+        ))}
+      </List.Section>
+    </List>
+  );
+}
+
+function useFuse<U>(
+  items: U[] | undefined,
+  options: Fuse.IFuseOptions<U>,
+  limit: number
+): [U[], Dispatch<SetStateAction<string>>] {
+  const [query, setQuery] = useState("");
+  const fuse = useMemo(() => {
+    return new Fuse(items || [], options);
+  }, [items]);
+
+  if (!query) return [(items || []).slice(0, limit), setQuery];
+  const results = fuse.search(query, { limit: limit });
+  return [results.map((result) => result.item), setQuery];
+}
+
+function EntryList(props: { doc: Doc; icon: ImageLike }) {
+  const { doc, icon } = props;
+  const { data, isLoading } = useFetchWithCache<{ entries: Entry[] }>(
+    `${DEVDOCS_BASE_URL}/docs/${doc.slug}/index.json`,
+    `${doc.slug}.json`
+  );
+  const [results, setQuery] = useFuse(data?.entries, { keys: ["name", "type"] }, 500);
+
+  return (
+    <List
+      isLoading={isLoading}
+      onSearchTextChange={(text) => {
+        setQuery(text);
+      }}
+    >
+      {results.map((entry) => (
+        <EntryItem entry={entry} icon={icon} key={entry.name + entry.path + entry.type} doc={doc} />
+      ))}
+    </List>
+  );
+}
+
+function EntryItem(props: { entry: Entry; doc: Doc; icon: ImageLike }) {
+  const { entry, doc, icon } = props;
+  return (
+    <List.Item
+      title={entry.name}
+      icon={icon}
+      key={entry.name + entry.path}
+      accessoryTitle={entry.type}
+      keywords={[entry.type].concat(entry.name.split("."))}
+      actions={
+        <ActionPanel>
+          <OpenInBrowserAction url={`${DEVDOCS_BASE_URL}/${doc.slug}/${entry.path}`} onOpen={() => popToRoot()} />
+          <OpenInDevdocsAction url={`${DEVDOCS_BASE_URL}/${doc.slug}/${entry.path}`} onOpen={() => popToRoot()} />
```

Comment:
Seems like they do the same?

Reply:
The second one open devdocs in the unofficial MacOS desktop app https://github.com/dteoh/devdocs-macos

Reply:
I added a redirection to the github page if the app is not currently installed.

Reply:
Ah, didn't see that one, my bad! Awesome addition though!